RecoTracker/MkFitCore/src/MkBuilder.cc

```diff
@@ -0,0 +1,1427 @@
+                                                   const int region,
+                                                   const int start_seed,
+                                                   const int itrack,
+                                                   const int end) {
+    // XXXX-1 If I miss a layer, insert the original track into tmp_cands
+    // AND do not do it in FindCandidates as the position can be badly
+    // screwed by then. See comment there, too.
+    // One could also do a pre-check ... so as not to use up a slot.
+
+    // bool debug = true;
+
+    for (int ti = itrack; ti < end; ++ti) {
+      TrackCand &cand = m_event_of_comb_cands[seed_cand_idx[ti].first][seed_cand_idx[ti].second];
+      WSR_Result &w = mkfndr->XWsrResult[ti - itrack];
+
+      // XXXX-4 Low pT tracks can miss a barrel layer ... and should be stopped
+      const float cand_r =
+          std::hypot(mkfndr->getPar(ti - itrack, MkBase::iP, 0), mkfndr->getPar(ti - itrack, MkBase::iP, 1));
+
+      dprintf("WSR Check label %d, seed %d, cand %d score %f -> wsr %d, in_gap %d\n",
+              cand.label(),
+              seed_cand_idx[ti].first,
+              seed_cand_idx[ti].second,
+              cand.score(),
+              w.m_wsr,
+              w.m_in_gap);
+
+      if (layer_info.is_barrel() && cand_r < layer_info.rin()) {
+        // Fake outside so it does not get processed in FindTracks Std/CE... and
+        // create a stopped replica in barrel and original copy if there is
+        // still chance to hit endcaps.
+        dprintf("Barrel cand propagated to r=%f ... layer is %f - %f\n", cand_r, layer_info.rin(), layer_info.rout());
+
+        mkfndr->XHitSize[ti - itrack] = 0;
+        w.m_wsr = WSR_Outside;
+
+        tmp_cands[seed_cand_idx[ti].first - start_seed].push_back(cand);
+        if (region == TrackerInfo::Reg_Barrel) {
+          dprintf(" creating extra stopped held back candidate\n");
+          tmp_cands[seed_cand_idx[ti].first - start_seed].back().addHitIdx(-2, layer_info.layer_id(), 0);
+        }
+      } else if (w.m_wsr == WSR_Outside) {
+        dprintf(" creating extra held back candidate\n");
+        tmp_cands[seed_cand_idx[ti].first - start_seed].push_back(cand);
+
+        // This can fire for Standard finding when candidates from a given seed are
+        // split between two iterations of the vecotrized loop over seeds as the
+        // output vector is shared between finding and the outside xtras here.
+        // if (tmp_cands[seed_cand_idx[ti].first - start_seed].size() > m_event_of_comb_cands[seed_cand_idx[ti].first].size())
+        //   printf("XXXXXXX %d %d for seed %d, index %d\n", (int) tmp_cands[seed_cand_idx[ti].first - start_seed].size(), (int) m_event_of_comb_cands[seed_cand_idx[ti].first].size(),
+        //          seed_cand_idx[ti].first, seed_cand_idx[ti].first - start_seed);
+      } else if (w.m_wsr == WSR_Edge) {
+        // XXXX-2 Additionally, if I miss/hit by epsilon, here would be a
```

Comment:
Could be moved to an issue?

Reply:
Moved to an internal issue.
Addressed in Round 4 follow-up PR trackreco/cmssw#79 commit: https://github.com/trackreco/cmssw/pull/79/commits/74f3afabd66bd5766e5129767b559c0888987831